ls1intum/Artemis
# `Development`: Improve input validation

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


In some places we don't validate the user's input properly.

### Description

I fixed the input validation where necessary and cleaned some sections up.

### Steps for Testing
We just want to make sure that I didn't accidentally break the implementation.

Prerequisites:
- 1 Instructor
- 1 Student

Course (Instructor/Student):
1. Look at a course with course image (e.g. 6.0 Testing Session Course 1 on TS1). Alternatively, create one
2. Make sure the image gets rendered correctly

After each step, commit and see if the changes were persisted by reloading
Programming Exercise (Instructor, Student):
1. Create a programming exercise with the online editor activated.
2. Change the programming language and make sure the template problem statement gets changed. Also change the application language and then the programming language and check if the natural language changes.
3. Participate in the programming exercise/alternatively edit the solution/template in the online editor
4. Create a folder
5. Create a folder within the folder
6. Create a file within the second folder
7. Rename the file
9. Delete the file and folders
10. No issues should occur

Fileupload Exercise (Instructor, Student):
1. Create a file upload exercise with a description. Upload an image to the description
2. Start the exercise, the image should show up properly as well as the rest of the description
3. Participate in the exercise
4. Upload another file
5. View/Download the image
6. No issues should occur

Quiz Exercise (Instructor):
1. Create a Quiz Exercise with a Drag and Drop Question including Background and image drag items
2. Preview the quiz and make sure all images get rendered properly

Split Lecture PDF (Instructor):
1. Open the lecture tab and click on "Create Lecture"
4. Fill in the title and channel name
5. Check "Automatic unit processing" at the bottom
7. Select a multipage PDF to process and hit "Process units". If you use a PDF that fits the requirements, skip point 5. Otherwise, continue.
8. Add units manually, at least two and set the pages accordingly. Then hit "submit.
9. Open the lecture and download a lecture unit attachment.
11. No issues should occur

#### Exam Mode Testing



We would need to test the participation checker which requires testing it on an app. However, the changes to the two endpoints are the exact same changes I did to a few other endpoints. Meaning if one fails, all should fail.
=> I deem it unnecessary to test every single of these endpoints. They already get tested by the other endpoints.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
